### ISSUE TITLE
Out-of-memory in Bremsstrahlung example 

### ISSUE THREAD

**Issue by n01r** [ComputationalRadiationPhysics/picongpu#1961]
I recently tried out the vanilla `Bremsstrahlung` example and noticed that `mallocMC` reports that it is out of memory.

The default in `cmakeFlags` in the example is 
```cmake
flags[0]="-DCUDA_ARCH=20"
```

I first used 8 k20 GPUs and the simulation was out of memory after 3000 time steps. By the same time the issues #1953 and #1954 came up. I double checked by compiling with 
```cmake
flags[0]="-DCUDA_ARCH=35"
```
and running on 8 k80 GPUs.

The simulation ran out of memory again and `stderr` is showing messages like:
```
[1,1]<stderr>:vsetenv LD_LIBRARY_PATH failed
...
[1,4]<stderr>:Send/receive buffer for species e is to small (max: 13797, direction: 2, retries: 1)
...
[1,3]<stderr>:  what():  [CUDA] Error: an illegal memory access was encountered[1,3]<stderr>:
[1,3]<stderr>:[kepler020:31195] *** Process received signal ***
[1,3]<stderr>:[kepler020:31195] Signal: Aborted (6)
[1,3]<stderr>:[kepler020:31195] Signal code:  (-6)
```
and long error messages pointing to memory exchanges.

Hopefully this is connected and can all be solved with #1960 .

**update** I forgot to post the module environment so here it is:
```
gcc/4.9.2
cmake/3.7.2
cuda/8.0
openmpi/1.8.6.kepler.cuda80
boost/1.60.0
pngwriter/0.5.6
hdf5-parallel/1.8.15
libsplash/1.6.0
numactl/2.0.7
paraview/3.98.laser
apr/1.5.1
apr-util/1.5.3
serf/1.3.6
svn/1.8.8
gnuplot/4.2.6
```

**Comment by ax3l** [comment]
@n01r please report if this was fixed with the latest updates :)

**Comment by n01r** [comment]
I ran the simulations mentioned above again but from a newly created example coming from the current `dev` (`commit 854318c252561ca5f3a372af5cce58213a43a574`) and with the same modules loaded.   

Unfortunately both runs crashed again after 60% and 90% of the set number of time steps in the default, respectively.

**Comment by PrometheusPi** [comment]
@n01r How many photons did the simulation create? Does using more GPUs and thus more memory avoid this issue? Do the last two runs still show the `illegal memory access` error and not an `terminate called after throwing an instance of 'std::runtime_error'
  what():  [CUDA] Error: out of memory` error?

**Comment by n01r** [comment]
@PrometheusPi that is the output from `ph_macroParticlesCount.dat`
```
#step count 
0 0 0.000000e+00
1000 1 1.000000e+00
2000 41818703 4.181870e+07
3000 383001600 3.830016e+08
4000 1138407498 1.138407e+09
```

[stderr.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/928415/stderr.txt)
[stdout.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/928417/stdout.txt)

I will try with more GPUs and also increase the `macroparticlesCount` output frequency

**Comment by n01r** [comment]
Due to the current occupation on our k80 queue I tried 32 k20 GPUs.   
The simulation finishes without any errors. So the default Bremsstrahlung example is just too large for the `0008gpus.cfg`. We still get a warning in `stderr`  that the send/receive buffer of mallocMC is too small, though.

I propose to 
1. Make the Bremsstrahlung example smaller / change the number of GPUs used by default.
2. Add `cmakeFlags` presets for `sm_35` and `sm_37`.

**Comment by ax3l** [comment]
Let's reduce the example to fit 8 GPUs if that is possible.

Keep the cmakeFlags with sm_20 if Bremsstrahlung does not explicitly need any newer hardware features, please.

**Comment by n01r** [comment]
Yep, I'd also rather reduce the example.
About the second point: I wouldn't remove the sm_20, I'd just add sm_35 and sm_37 as non-default presets. Does it bring any value to have these architectures tested as well when the compile suite tries out all the different presets, @ax3l ?

**Comment by ax3l** [comment]
Unfortunately it brings no additional value to add those SMs.

**Comment by n01r** [comment]
I tried a few things to reduce the number of photons in this example.
What I found to be most effective, was to reduce the value of `WEIGHTING_RATIO` in `bremsstrahlung.param`. This parameter is the ratio between electron weighting and photon weighting. The weighting of the newly created photons is calculated from it - but also the emission probability is influenced by it. If one decreases `WEIGHTING_RATIO` also the emission probability in `Bremsstrahlung.tpp` is reduced. That in turn decreases the photon statistics quality as it reduces the overall number of photons.
